### DISCUSSION TITLE
'./configure --prefix=/usr' put /etc directory in /usr/etc

### DISCUSSION THREAD

**Discussion by sputnick-dev** [HeyuX10Automation/heyu#57] [Q&A] [answered]
```
$ wget https://github.com/HeyuX10Automation/heyu/archive/v2.11-rc2.tar.gz
$ tar zxvf archive/v2.11-rc2.tar.gz
$ cd heyu-2.11-rc2/
$ ./configure --prefix=/tmp/usr
$ make
$ make install
$ find /tmp/usr
/tmp/usr
/tmp/usr/etc
/tmp/usr/etc/heyu
/tmp/usr/etc/heyu/x10.sched.sample
/tmp/usr/etc/heyu/x10config.sample
/tmp/usr/share
...
```

As you can see, /etc should not be in */usr/etc but /etc

(/tmp is there for debug purpose, feel free to strip it)

**Comment by jkrzyszt** [comment, marked as answer]
Please try:
`./configure --prefix=/tmp/usr --sysconfdir=/etc`
You can also run `sh ./Configure.sh` instead.
More info can be found in `README` file.

Thanks,
Janusz

**Comment by sputnick-dev** [comment]
Ok, it works, but this is the first time (for 10 years as dev/sysadmin) I see an application that install /etc files elsewhere than /etc. Seems not standard...

**Comment by jkrzyszt** [comment]
Please have a look at this discussion:
https://lists.gnu.org/archive/html/automake/2007-01/msg00069.html

Thanks,
Janusz

**Comment by sputnick-dev** [comment]
AFAIK, this seems quite non standard. 

To be able to have each part where it should, I finally wrote : 

``` bash
  ./configure --localstatedir=/var --mandir=/usr/share/man --enable-postinst=./post-install.sh --sysconfdir=/etc --prefix=/usr
  make
```

https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=heyu

**Comment by jkrzyszt** [comment]
> AFAIK, this seems quite non standard.

If still not convinced, please have a look at the relevant section of GNU Coding Standards document:
https://www.gnu.org/prep/standards/standards.html#Directory-Variables

> ```
> ./configure --localstatedir=/var --mandir=/usr/share/man --enable-postinst=./post-install.sh --sysconfdir=/etc --prefix=/usr
> make
> ```

That's pretty similar to what `sh Configure.sh --prefix=/usr` would do for Linux.

> https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=heyu

Thanks a lot for packaging Heyu for Arch Linux !!!
Janusz

**Comment by jkrzyszt** [comment]
Correcting myself:

> > ```
> > ./configure --localstatedir=/var --mandir=/usr/share/man --enable-postinst=./post-install.sh --sysconfdir=/etc --prefix=/usr
> > make
> > ```
> 
> That's pretty similar to what `sh Configure.sh --prefix=/usr` would do for Linux.

That actually doesn't work that way, sorry. `Configure.sh` itself doesn't accept Autoconf specific options. Being a wrapper over `configure`, it mimics behaviour of Heyu historical (pre-Autoconf) `Configure` script. For Linux that means `/usr/local` prefix is used.

Either use `Configure.sh` with traditional Heyu directory layout, or fine tune it up to your needs by calling `configure` with your preferred options directly.

Thanks,
Janusz
